### DISCUSSION TITLE
https://angular.love/angular-dependency-inversion-principle-2

### DISCUSSION THREAD

**Discussion by giscus** [HouseOfAngular/angular-love#365] [General]
# https://angular.love/angular-dependency-inversion-principle-2

Angular.love - a place for all Angular enthusiasts created to inspire and educate.

https://angular.love/angular-dependency-inversion-principle-2

**Comment by EnverUsta** [comment]
I have confusion here. Our initial motivation was getting rid of if clauses. Hıwever, in the end if we don't know what we need in the module level, we needed to implenent a factory which has if clauses(switch cases in your example). Therefore, what is the benefit of this approach in that case? Maybe more structural implementation?
